### PR TITLE
Add name function to actor trait

### PR DESCRIPTION
**Pull request by jeffa5**
This enables showing a useful name in the explorer next to the id, e.g. differentiating servers and clients more clearly.

I'd be happy with updating the `name` function's args if needed.

### CODE REVIEW

**Comment by jeffa5** [comment]
Example with the paxos example:
image

**Comment by jonnadal** [comment]
Great idea! Can't wait to try this. Will follow up later in the week.

File: src/actor/model.rs [hunk 1]
```diff
@@ -785,7 +810,8 @@ mod test {
                 .spawn_bfs()
                 .join()
                 .unique_state_count(),
-            2); // initial and delivery of Interesting
+            2
+        ); // initial and delivery of Interesting
```
Comment (jonnadal) [inline src/actor/model.rs:814]: Not a blocker: comment is no longer associated w/ the line it references (and there are some other examples). Leaving this here as a reminder to correct later.

File: src/actor/register.rs [hunk 2]
```diff
@@ -127,6 +127,23 @@ where
     type State = RegisterActorState<ServerActor::State, u64>;
     type Timer = ServerActor::Timer;
 
+    fn name(&self) -> String {
+        match self {
+            RegisterActor::Client {
+                put_count: _,
+                server_count: _,
+            } => "Client".to_owned(),
```
Comment (jonnadal) [inline src/actor/register.rs:135]: Reminder for fix-up later: I think `{ .. }` syntax would suffice.

**Review by jonnadal — APPROVED** [review]

**Comment by jonnadal** [comment]
This is a such a good addition! Thank you once again @jeffa5.

File: src/actor/model.rs (hunk 1 above)
Comment (jeffa5) [reply src/actor/model.rs:814]: Done in https://github.com/stateright/stateright/commit/1466b3ca650a53c6910107ae95500e9664c8ad50

File: src/actor/register.rs (hunk 2 above)
Comment (jeffa5) [reply src/actor/register.rs:135]: Done in https://github.com/stateright/stateright/commit/1466b3ca650a53c6910107ae95500e9664c8ad50

**Pull request merged**